src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -331,6 +345,20 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun pluralise(count: Int, singular: String, plural: String): String =
         if (count == 1) singular else plural
 
+    private fun canElideReturnLifetime(fn: PsiElement): Boolean {
```

Comment:
I think this can work, provided that we gracefully handle the case where we fail to resolve. Would be interesting to find some documentation/rustc source code which precisely specifies how lifetimes can be elided. 

We can also scale this PR to fields, which are handled correctly, and solve lifetime elision separately.